### ISSUE TITLE
Not able to configure/install the required packages.

### ISSUE THREAD

**Issue by itsyogesh** [stephenmathieson-boneyard/node-sparkey#5]
I am trying to do a npm install for node-gyp to rebuild the whole project, but it throws an error. Here is the stack trace of the error.

make: Entering directory `/home/yogeshkumar/dbTests/node-sparkey/build'
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy-sinksource.o
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy-stubs-internal.o
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy.o
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.1/snappy-c.o
  AR(target) Release/obj.target/deps/snappy/snappy.a
  COPY Release/snappy.a
make: *** No rule to make target`Release/obj.target/sparkey/deps/sparkey/src/buf.o', needed by `Release/obj.target/deps/sparkey.a'.  Stop.
make: Leaving directory`/home/yogeshkumar/dbTests/node-sparkey/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/yogeshkumar/.linuxbrew/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:810:12)
gyp ERR! System Linux 3.5.0-59-generic
gyp ERR! command "node" "/home/yogeshkumar/.linuxbrew/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/yogeshkumar/dbTests/node-sparkey
gyp ERR! node -v v0.10.33
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok 

npm ERR! Linux 3.5.0-59-generic
npm ERR! argv "node" "/home/yogeshkumar/.linuxbrew/bin/npm" "install"
npm ERR! node v0.10.33
npm ERR! npm  v2.1.6
npm ERR! code ELIFECYCLE
npm ERR! sparkey@0.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the sparkey@0.1.0 install script.
npm ERR! This is most likely a problem with the sparkey package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls sparkey
npm ERR! There is likely additional logging output above.

**Comment by itsyogesh** [comment]
Sorry, I figured out what the error was. I didn't copy the source files of sparkey in the deps folders. Its working now. Thanks.

**Comment by stephenmathieson** [comment]
ah yeah, if you don't `npm install sparkey`, you'll have to pull down the sparkey submodule too.

**Comment by stephenmathieson** [comment]
btw, @yogeshkumar180592 - this module is not actively maintained.  it was a POC for something i was working on about a year ago and i decided to go with a different k/v store.

**Comment by itsyogesh** [comment]
Whoa, that was a really good POC. So what k/v store did you finally end up using. I needed something with a very high read rate, and the sparkey docs say that it is made for this job. But setting up and configuring everything is a real pain, mostly because of no getting started guides. Do you have any recommendations for a different k/v store or should I stick with this for the time being.

**Comment by stephenmathieson** [comment]
thanks! :)

i stuck with [Sophia](https://github.com/pmwkaa/sophia) and am using [Sophist](https://github.com/stephenmathieson/node-sophist) in our HTTP app.  we've got a native C app (with Sophia embedded directly) which writes locally and pushes bulk data up semi-frequently to sync everything together.  it's a bit convoluted as Sophia 1.1 doesn't support multiple transactions at once, so we built a queue around everything.

in order to recommend a k/v store for you, i'd need to know a bit more about what you're trying to accomplish.  maybe shoot me an email?
